src/dialog/unix/SDL_portaldialog.c

```diff
@@ -234,7 +236,12 @@ static DBusHandlerResult DBus_MessageFilter(DBusConnection *conn, DBusMessage *m
                     goto cleanup;
                 }
             }
-            dbus->message_iter_get_basic(&uri_entry, path + current);
+
+            dbus->message_iter_get_basic(&uri_entry, &uri);
+
+            /* https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.FileChooser.html */
+            /* Returned paths will always start with 'file://'; truncate it */
+            path[current] = uri + SDL_strlen("file://");
```

Comment:
You should check whether the uri starts with that prefix first. Future versions of the protocol may return other URI formats.

Reply:
I've also added some code to check if there's another protocol, and to report it as unsupported if so.